Include packages collected by scancode in reports

yannjor opened the pull request:
Currently only file data is included when tern is run with the scancode
extension, even though scancode can also detect packages. These changes
add the package information collected by scancode to tern's data model.

Resolves #790

Signed-off-by: Yann Jorelle <yann.jorelle@nokia.com>

nishakm commented:
@yannjor Thanks for the PR! The changes look good to me. I noticed that there were no changes on reconciling packages inventoried by different tools. Do you have any thoughts on that?

yannjor commented:
> I noticed that there were no changes on reconciling packages inventoried by different tools. Do you have any thoughts on that?

@nishakm I had not thought about that :blush: I added one commit that tries to do this. It works so that if a package detected by scancode was already found some other way, it tries to fill in missing metadata in the existing one with the metadata of the one detected by scancode. I added a merge method to the Package class similar to the one in FileData. Do you agree with this approach?

nishakm commented:
> > I noticed that there were no changes on reconciling packages inventoried by different tools. Do you have any thoughts on that?
> 
> @nishakm I had not thought about that 😊 I added one commit that tries to do this. It works so that if a package detected by scancode was already found some other way, it tries to fill in missing metadata in the existing one with the metadata of the one detected by scancode. I added a merge method to the Package class similar to the one in FileData. Do you agree with this approach?

I actually forgot FileData had a merge method. That will do for now. Thanks so much!

nishakm commented:
Tested it on an nginx container. Looks good! Merging.
